tests/tools/pytest_utils.py

```diff
@@ -166,13 +166,19 @@ def _chunk_graph(
     metadata["edges"] = {}
     with setdir("edge_index"):
         np.random.shuffle(idxes)
+        valid_delimiters = [" ", ",", "\t", ":"]
         for idx in idxes:
             etype = idxes_etypestr[idx][0]
             etypestr = idxes_etypestr[idx][1]
             logging.info("Chunking edge index for %s" % etypestr)
             edges_meta = {}
             if edges_format == "csv":
-                fmt_meta = {"name": edges_format, "delimiter": " "}
+                fmt_meta = {
+                    "name": edges_format,
+                    "delimiter": valid_delimiters[
+                        np.random.randint(0, len(valid_delimiters))
```

Comment:
Won't this result in a potentially different outcome every time the test is executed? You want tests to be reproducible and this is not. 

If you want to test all possible delimiters add them as a parametrized test.

In regards to overall test design:
This whole process reads more like an integration test that tests multiple functions together. A good idea is to have isolated tests for functional units.

Reply:
Agreed this code snippet is backed out now.